civo/instances/resource_instance.go

```diff
@@ -190,6 +190,19 @@ func ResourceInstance() *schema.Resource {
 func resourceInstanceCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	apiClient := m.(*civogo.Client)
 
+	hostname := d.Get("hostname").(string)
+
+	// Check if an instance already exists by this hostname
+	existingInstances, err := apiClient.ListAllInstances()
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
Redundant API call. We could just check for AlreadyExist error from the SDK.

Reply:
Yes, it is one extra API call. I get it. Thank you @uzaxirr